### ISSUE TITLE
"%x format: an integer is required, not gdb.Value" on latest gdb/python3

### ISSUE THREAD

**Issue by gsingh93** [cloudburst/libheap#6]
On Arch Linux I'm getting the following error when running `heap`:
```
Python Exception <class 'TypeError'> %x format: an integer is required, not gdb.Value: 
Error occurred in Python command: %x format: an integer is required, not gdb.Value
```

Debugging showed this line as throwing the exception:
```
print("\t arena @ 0x%x" % \
                        ar_ptr.address.cast(gdb.lookup_type("unsigned long")))
```
(the exception handler throws the same exception)

I've used this library before and this wasn't happening, so I think it happened when either gdb (version 7.11) or python3 (version 3.5.1) updated. I found the fix [here](https://sourceware.org/ml/gdb-patches/2016-01/msg00409.html), which is to wrap the `gdb.Value` in a call to `int`.

**Comment by cloudburst** [comment]
Thanks!  I don't have a python3 environment to test a patch on but I will fix this in the future.

**Comment by gsingh93** [comment]
I'll do my best to make a PR in the next few days (I've fixed it somewhat locally), but I probably won't cover every case, and there are still a few issues I couldn't fix, but I'll describe all this in the PR.

**Comment by cloudburst** [comment]
I pulled the fixes out of your fork and tried to catch the rest.  Hopefully fixed in ca0f5ccd31753efd0b613b8a2575d1369a9e2466.  Sorry I typoed your name in the commit log.  Please reopen if you see this issue anywhere else.
